cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -142,6 +142,65 @@ using rhs_iterator = strong_index_iterator<rhs_index_type>;
 
 namespace lexicographic {
 
+/**
+ * @brief Relational comparator functor that compares physical values rather than logical
+ * elements like lists, strings, or structs. It evaluates `NaN` as not less than, equal to, or
+ * greater than other values and is IEEE-754 compliant.
+ */
+struct physical_element_comparator {
+  /**
+   * @brief Operator for relational comparisons.
+   *
+   * @param lhs First element
+   * @param rhs Second element
+   * @return Relation between elements
+   */
+  template <typename Element>
+  __device__ constexpr weak_ordering operator()(Element const lhs, Element const rhs) const noexcept
+  {
+    return detail::compare_elements(lhs, rhs);
+  }
+};
```

Comment:
I'm noticing that this isn't used outside of testing. While I understand why we have it (largely because I probably asked for it) I don't think we need it because it's not useful as a base class like I originally thought.